syntax = "proto3";
package main;
option go_package="./;mygrpc";


message ResStatus {
    int32 statusCode = 1;
    string errCode = 2;
    string errMessage = 3;
}

message ResEmpty {
    ResStatus status = 1;
}

message ReqEmpty {
}

message ReqMLResult {
    string ContainerID = 1;
}

message ResMLResult {
    ResStatus status = 1;
    string ContainerID = 2;
}

message ReqConfirmContainerID {
    string ContainerID = 1;
}

message ResConfirmContainerID {
    ResStatus status = 1;
}

service MyGRPC {
    rpc newMLResult(ReqMLResult) returns (ResEmpty);
    rpc pullMLResult(ReqEmpty) returns (stream ResMLResult);
    rpc confirmContainerID(ReqConfirmContainerID) returns (ResConfirmContainerID);
}